### ISSUE TITLE
spawnChildProcess args bug

### ISSUE THREAD

**Issue by squarewav** [asamuzaK/withExEditorHost#66]
Hi,

When repeatedly launching and closing the external editor, the path to the temp file is indirectly added to the editotrConfig.cmdArgs each time. This causes the args variable to accumulate extra redundant path arguments. When the external editor is vim, this causes vim to report an error like:

`E173: 1 more file to edit`

And the path is the old path so the actual content can become incorrect.

The path is added by spawn() to the cmdArgs indirectly through the args reference.

You can see this by monitoring the args variable and repeatedly launching and closing the external editor multiple times. When logging args it looks like this:

`-e,vim,/tmp/withExEditor/28767/tmpFiles/1/1016/taj/html_input_3.txt,/tmp/withExEditor/28767/tmpFiles/1/1016/taj/html_input_3.txt,/tmp/withExEditor/28767/tmpFiles/1/1016/taj/html_input_2.txt`

The fix is fairly simple. Just add a slice() call to effectively copy the args array and break the reference:

>   }
>   const args = (new CmdArgs(editorConfig.cmdArgs)).toArray()**.slice()**;
>   const opt = {
>     cwd: null,

**Comment by squarewav** [comment]
Also, on a somewhat unrelated note, the documentation for spawn claims that you should be calling spawn more like this:

```
let args = editorConfig.cmdArgs.slice();
args.push(file);
const spawn = require('child_process').spawn;
const proc = await spawn(app, args);
```

and not instantiate ChildProcess directly.

**Comment by asamuzaK** [comment]
Confirmed.
Thanks for the bug report.

**Comment by asamuzaK** [comment]
Released v5.1.7
